zone.go

```diff
@@ -444,6 +451,43 @@ func (api *API) ListZonesContext(ctx context.Context, opts ...ReqOption) (r Zone
 		return ZonesResponse{}, errors.Wrap(err, errUnmarshalError)
 	}
 
+	totalPageCount := r.TotalPages
+	var wg sync.WaitGroup
+	wg.Add(totalPageCount)
+	errc := make(chan error)
```

Comment:
It doesn't seem we handle errors returned from channel in any way. We should return them at the end of the function, even a single one is enough. We might also want to cancel pending requests if any goroutine fails this way.

Reply:
Is [this](https://github.com/cloudflare/cloudflare-go/pull/534/files#diff-6d1d02aea42947cbb0ae78b9690828a2R477-R478) what you mean? Or did you have something else in mind for the error being returned? I'm not explicitly cancelling the routine but can drop it in if the channel error reporting isn't enough here.

Reply:
Circling back on this one @patryk as we've still got an issue open in the Terraform provider depending on this one.